akvo/rsr/filters.py

```diff
@@ -98,26 +98,45 @@ def get_locations(location, locations):
 def location_choices(qs):
     """From a queryset get possible location filter choices"""
 
-    if qs.model is Project:
-        location_model = ProjectLocation
-    elif qs.model is ProjectUpdate:
+    country_ids = (
+        get_recipient_country_ids(qs) if qs.model is Project else
```

Comment:
I ran this query (WARNING it's a bomb, takes almost three minutes to run the query on my vagrant!):

```sql
SELECT p.id
FROM rsr_project p, rsr_publishingstatus s
WHERE p.id NOT IN (SELECT DISTINCT p.id
                   FROM rsr_project p, rsr_recipientcountry r, rsr_projectlocation l
                   WHERE p.id = r.project_id OR p.id = l.location_target_id) AND p.id = s.project_id
      AND s.status = 'published'
```

 on a month old RSR DB but that's had the country-from-coords migration applied. 41 published projects have no country at all 😢